Assets/BowAndArrowImproved/Scripts/Enemies/BAAIEnemy.cs

```diff
@@ -29,15 +32,20 @@ private void OnEnable()
         PlayWarCry();
     }
 
-    public void Die(bool playSound)
+    public void Die(bool killedByPlayer)
     {
         gameObject.SetActive(false);
 
-        if (playSound)
+        if (killedByPlayer)
         {
-            AudioSource.PlayClipAtPoint(breakClips[Random.Range(0, breakClips.Count)], transform.position);    
+            AudioSource.PlayClipAtPoint(breakClips[Random.Range(0, breakClips.Count)], transform.position);
+            UpdateScoreOnDeath updateScoreOnDeathScript = gameObject.GetComponent<UpdateScoreOnDeath>();
+            if (updateScoreOnDeathScript != null)
+            {
+                updateScoreOnDeathScript.isUpdateValid = true;
+            }
```

Comment:
bad code, nobody cares about this script here - pass another argument to the event